### PR TITLE
failing test for streams issue

### PR DESCRIPTION
**Pull request by raytiley**
This recreates #10539 in a unit test

Thanks @mmun for helping out. I'm going to keep digging into this.

### CODE REVIEW

File: packages/ember-views/tests/views/view/stream_test.js [hunk 1]
```diff
@@ -31,6 +31,44 @@ QUnit.test("can return a stream that is notified of changes", function() {
...
+
+  var stream = view.getStream('view.foo.bar');
+  stream.value();
+  stream.subscribe(function() {
+    stream.value();
+    count++;
+  });
+
+  // run
+  view.set('foo', { bar: 'mmun' });
```
Comment (fivetanley) [inline packages/ember-views/tests/views/view/stream_test.js]: I think these might need to be wrapped in ember.run. Looks like `run` is used in other tests.

Comment (mmun) [inline packages/ember-views/tests/views/view/stream_test.js]: There's nothing in this test that schedules on the run loop. The other tests use a template and template bindings, so they do need ember.run

File: packages/ember-views/tests/views/view/stream_test.js [hunk 2]
```diff
@@ -31,6 +31,44 @@ QUnit.test("can return a stream that is notified of changes", function() {
...
+  view.set('foo', { bar: 'bantic' });
+  equal(view.get('foo.bar'), 'bantic');
+
+  view.set('foo.bar', 'mixonic');
+  equal(view.get('foo.bar'), 'mixonic');
+
+  view.set('foo.bar', 'teddyzeenny');
+  equal(view.get('foo.bar'), 'teddyzeenny');
+
+  equal(count, 6, 'fires subscriber twice');
```
Comment (mmun) [inline packages/ember-views/tests/views/view/stream_test.js]: "Fires subcribers correct number of times."

File: packages/ember-views/lib/streams/key_stream.js [hunk 3]
```diff
@@ -38,6 +38,10 @@ merge(KeyStream.prototype, {
     if (nextObj !== prevObj) {
       if (prevObj && typeof prevObj === 'object') {
         removeObserver(prevObj, this.key, this, this.notify);
+
+        if (nextObj && typeof nextObj === 'object') {
```
Comment (mmun) [inline packages/ember-views/lib/streams/key_stream.js]: This should be moved out of the conditional (to handle the case when `prevObj` is `null`).

Comment (raytiley) [inline packages/ember-views/lib/streams/key_stream.js]: I had that originally but it causes this assertion to fail: https://github.com/emberjs/ember.js/blob/master/packages/ember-views/tests/views/view/class_name_bindings_test.js#L214

**Pull request closed without merging**